chia/wallet/did_wallet/did_wallet.py

```diff
@@ -203,16 +204,18 @@ async def create_new_did_wallet_from_coin_spend(
         args = did_wallet_puzzles.uncurry_innerpuz(inner_puzzle)
         if args is None:
             raise ValueError("Cannot uncurry the DID puzzle.")
-        _, _, num_verification, _, metadata = args
-        # full_solution: Program = Program.from_bytes(bytes(coin_spend.solution))
-        # inner_solution: Program = full_solution.rest().rest().first()
+        _, recovery_list_hash, num_verification, _, metadata = args
+        full_solution: Program = Program.from_bytes(bytes(coin_spend.solution))
+        inner_solution: Program = full_solution.rest().rest().first()
         recovery_list: List[bytes] = []
-        # for did in list(inner_solution.rest().rest().rest().rest().rest().rest().as_iter()):
-        #     recovery_list.append(did.as_python()[0])
+        backup_required: int = num_verification.as_int()
+        if recovery_list_hash != Program.to([]).get_tree_hash():
+            for did in list(inner_solution.rest().rest().rest().rest().rest().as_iter()):
```

Comment:
What does the wallet do if the reveal is not present?

Reply:
It will reset recovery list to [] and the required number to 0